crates/yttrium/src/chain_abstraction/client.rs

```diff
@@ -67,4 +80,57 @@ impl Client {
             Err(RouteError::RequestFailed(response.text().await))
         }
     }
+    pub async fn wait_for_success(
+        &self,
+        orchestration_id: String,
+        check_in: Duration,
+    ) -> Result<StatusResponseCompleted, WaitForSuccessError> {
+        self.wait_for_success_with_timeout(
+            orchestration_id,
+            check_in,
+            Duration::from_secs(30),
+        )
+        .await
+    }
+
+    pub async fn wait_for_success_with_timeout(
```

Comment:
Why do we need `wait_for_success` and `wait_for_success_with_timeout`?
Doesn't BC API timeout itself?

Reply:
This timeout is how long the client will poll the Blockchain API's `/status` endpoint

Reply:
I get it but why client should specify that number? client doesn't know it and API should time out itself

Reply:
I think we still need a timeout here to handle server issues where what you suggest isn't implemented or is implemented incorrectly.

[Slack conversation](https://reown-inc.slack.com/archives/C0816SK4877/p1732211619422449) for establishing a server-side timeout